src/server/apsystem.ts

```diff
@@ -539,12 +539,20 @@ export default class ActivityPubSystem {
         )
         return
       }
+
       // Remove activity from inbox and any other index/collection
-      if (typeof activity.object !== 'string') {
-        throw createError(400, 'Error deleting activity, must have activity URL in object field')
+      let object = activity.object
+      if (typeof object !== 'string') {
+        if ((object != null) && 'id' in object && typeof object.id === 'string') {
+          object = object.id
```

Comment:
can we have a helper to obtain the object id?

Reply:
Yeah I think we could stand to have a bunch of helpers, not sure when to fit it in